src/controller/authController.js

```diff
@@ -0,0 +1,97 @@
+import bcrypt from "bcrypt";
+import Jwt from "../utils/jwt.js";
+import { getUserByEmail, registerGoogle } from "../services/user.services.js";
+import db from "../database/models/index.js";
+
+export const googleAuthHandler = async (req, res) => {
+  const { value } = req.user.emails[0];
+  const { familyName } = req.user.name;
+  const newUser = {
+    name: familyName,
+    email: value,
+    password: "12345",
+  };
+
+  // Check if user already exists
+  const user = await getUserByEmail(newUser.email);
+  if (user) {
+    // User already exists, return response as json
+    const { id, email, name, password } = user;
+    const userToken = Jwt.generateToken(
+      {
+        id: id,
+        email: email,
+        name: name,
+        password: password,
+      },
+      "1h"
+    );
+  }
+  // Hash the password before registering the user
+  const saltRounds = 10;
+  const hashedPassword = await bcrypt.hash(newUser.password, saltRounds);
+  newUser.password = hashedPassword;
+
+  const { id, email, name, password } = await registerGoogle(newUser);
+  const userToken = Jwt.generateToken(
+    {
+      id: id,
+      email: email,
+      name: name,
+      password: password,
+    },
+    "1h"
+  );
+  return res.redirect(`/api/callback?key=${userToken}`);
+};
+// get the user from the database
+
+export const GetUsers = async (req, res) => {
+  try {
+    const users = await db.User.findAll();
+    return res.status(200).json({ users });
```

Comment:
Let's use Jsend format to structure our responses.

Reply:
done!